pkg/verification/verification_service_test.go

```diff
@@ -150,6 +159,211 @@ func (s *TestVerificationServiceSuite) TestSendVerifyMessageFails() {
+					v1alpha1.UserVerificationAttemptsAnnotationKey:        "3",
+					v1alpha1.UserSignupVerificationCodeAnnotationKey:      "123456",
+					v1alpha1.UserVerificationExpiryAnnotationKey:          now.Add(10 * time.Second).Format(verification.TimestampLayout),
+				},
+				Labels: map[string]string{
+					v1alpha1.UserSignupPhoneNumberLabelKey: "+1NUMBER",
+				},
+			},
+			Spec: v1alpha1.UserSignupSpec{
+				Username: "sbryzak@redhat.com",
+			},
+		}
+
+		err := svc.VerifyCode(ctx, userSignup, "123456")
+		require.Error(s.T(), err)
+		require.Equal(s.T(), "Too many requests: too many verification attempts", err.Error())
+	})
+
+	s.T().Run("when verifications attempts has invalid value", func(t *testing.T) {
+
+		userSignup := &v1alpha1.UserSignup{
+			TypeMeta: v1.TypeMeta{},
+			ObjectMeta: v1.ObjectMeta{
+				Name:      "123",
+				Namespace: "test",
+				Annotations: map[string]string{
+					v1alpha1.UserSignupUserEmailAnnotationKey:             "sbryzak@redhat.com",
+					v1alpha1.UserSignupVerificationTimestampAnnotationKey: now.Add(-1 * time.Minute).Format(verification.TimestampLayout),
+					v1alpha1.UserVerificationAttemptsAnnotationKey:        "ABC",
+					v1alpha1.UserSignupVerificationCodeAnnotationKey:      "123456",
+					v1alpha1.UserVerificationExpiryAnnotationKey:          now.Add(10 * time.Second).Format(verification.TimestampLayout),
+				},
+				Labels: map[string]string{
+					v1alpha1.UserSignupPhoneNumberLabelKey: "+1NUMBER",
+				},
+			},
+			Spec: v1alpha1.UserSignupSpec{
+				Username: "sbryzak@redhat.com",
+			},
+		}
+
+		err := svc.VerifyCode(ctx, userSignup, "123456")
+		require.Error(s.T(), err)
+		require.Equal(s.T(), "strconv.Atoi: parsing \"ABC\": invalid syntax", err.Error())
+		require.Equal(s.T(), "3", userSignup.Annotations[v1alpha1.UserVerificationAttemptsAnnotationKey])
+	})
+
+	s.T().Run("when verifications expiry is corrupt", func(t *testing.T) {
+
+		userSignup := &v1alpha1.UserSignup{
+			TypeMeta: v1.TypeMeta{},
+			ObjectMeta: v1.ObjectMeta{
+				Name:      "123",
+				Namespace: "test",
+				Annotations: map[string]string{
+					v1alpha1.UserSignupUserEmailAnnotationKey:             "sbryzak@redhat.com",
+					v1alpha1.UserSignupVerificationTimestampAnnotationKey: now.Add(-1 * time.Minute).Format(verification.TimestampLayout),
+					v1alpha1.UserVerificationAttemptsAnnotationKey:        "0",
+					v1alpha1.UserSignupVerificationCodeAnnotationKey:      "123456",
+					v1alpha1.UserVerificationExpiryAnnotationKey:          "ABC",
+				},
+				Labels: map[string]string{
+					v1alpha1.UserSignupPhoneNumberLabelKey: "+1NUMBER",
+				},
+			},
+			Spec: v1alpha1.UserSignupSpec{
+				Username: "sbryzak@redhat.com",
+			},
+		}
+
+		err := svc.VerifyCode(ctx, userSignup, "123456")
+		require.Error(s.T(), err)
+		require.Equal(s.T(), "Internal error occurred: parsing time \"ABC\" as \"2006-01-02T15:04:05.000Z07:00\": cannot parse \"ABC\" as \"2006\"", err.Error())
+	})
+}
+
```

Comment:
I guess we could move some code here to reusable functions too?

Reply:
See my other comment above.  Also, there's quite a variation of UserSignups that are created, some of them only need the email and phone number, others need all of the verification annotations.